pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller_test.go

```diff
@@ -1051,6 +1055,90 @@ func TestDeployOnKube111(t *testing.T) {
 	}
 }
 
+func TestRenderingDefaultResourceRequirements(t *testing.T) {
+	r := &operatorapiv1.ResourceRequirement{
+		Type: operatorapiv1.ResourceQosClassDefault,
+	}
+	config := newFakeKlusterletConfigWithResourceRequirement(t, r)
+	for _, file := range getManifestFiles() {
+		manifest, err := manifests.KlusterletManifestFiles.ReadFile(file)
+		if err != nil {
+			t.Errorf("Failed to read file %s", file)
+		}
+		objData := assets.MustCreateAssetFromTemplate(file, manifest, config).Data
+		deploy := &appsv1.Deployment{}
+		if err = yaml.Unmarshal(objData, deploy); err != nil {
+			t.Errorf("Failed to unmarshal deployment: %v", err)
+		}
+		requests := deploy.Spec.Template.Spec.Containers[0].Resources.Requests
+		actualCpu := requests.Cpu().String()
+		if actualCpu == "" {
+			t.Errorf("requests.cpu should not be empty")
+		}
+		actualMem := requests.Memory().String()
+		if actualMem == "" {
+			t.Errorf("requests.memory should not be empty")
+		}
+	}
+}
+
+func TestRenderingBestEffortResourceRequirements(t *testing.T) {
+	r := &operatorapiv1.ResourceRequirement{
+		Type: operatorapiv1.ResourceQosClassBestEffort,
+	}
+	config := newFakeKlusterletConfigWithResourceRequirement(t, r)
+	for _, file := range getManifestFiles() {
+		manifest, err := manifests.KlusterletManifestFiles.ReadFile(file)
+		if err != nil {
+			t.Errorf("Failed to read file %s", file)
+		}
+		objData := assets.MustCreateAssetFromTemplate(file, manifest, config).Data
+		deploy := &appsv1.Deployment{}
+		if err = yaml.Unmarshal(objData, deploy); err != nil {
+			t.Errorf("Failed to unmarshal deployment: %v", err)
+		}
+		actualResources := deploy.Spec.Template.Spec.Containers[0].Resources.String()
+		req := corev1.ResourceRequirements{}
+		actualRequests := req.String()
+		if actualResources != actualRequests {
+			t.Errorf("expect:\n%s\nbut got:\n%s", actualRequests, actualResources)
+		}
+	}
+}
+
+func TestRenderingResourceRequirements(t *testing.T) {
+	r := &operatorapiv1.ResourceRequirement{
```

Comment:
seems to me the three tests can be put in one test func with different cases. It also worth a test case with type is ResourceRequirement and ResourceRequirements is nil or empty

Reply:
merged into one unit test with 3 cases